mirascope/core/anthropic/call_response.py

```diff
@@ -78,17 +94,14 @@ def tools(self) -> list[AnthropicTool] | None:
         Raises:
             ValidationError: if a tool call doesn't match the tool's schema.
         """
-        if not self.tool_types:
+        if not self.tool_types or not self.tool_calls:
             return None
 
         extracted_tools = []
-        for tool_call in self.response.content:
-            if tool_call.type != "tool_use":
-                continue
+        for tool_call in self.tool_calls:
```

Comment:
this doubles the iteration since we iterate first to get the tool calls and then again through the tool calls. I prefer having just a single iteration here

Reply:
reverted